$MirOS: ports/security/gnupg/patches/patch-util_secmem_c,v 1.1 2006/09/11 01:02:04 tg Exp $
--- util/secmem.c.orig	Tue Oct 23 07:55:08 2007
+++ util/secmem.c	Tue Apr  1 15:22:25 2008
@@ -36,6 +36,15 @@
 #include <sys/lock.h>
 #endif
 #endif
+#ifdef HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif
+#ifdef HAVE_SYS_SYSCTL_H
+#include <sys/sysctl.h>
+#endif
+#ifdef HAVE_UVM_UVM_SWAP_ENCRYPT_H
+#include <uvm/uvm_swap_encrypt.h>
+#endif
 
 #include "types.h"
 #include "memory.h"
@@ -91,6 +100,19 @@ static int suspend_warning;
 static void
 print_warn(void)
 {
+#if defined(HAVE_SYS_PARAM_H) && defined(HAVE_SYS_SYSCTL_H) && \
+    defined(CTL_VM) && defined(VM_SWAPENCRYPT) && defined(SWPENC_ENABLE)
+  int mib[3], swapenc;
+  size_t len;
+
+  mib[0] = CTL_VM;
+  mib[1] = VM_SWAPENCRYPT;
+  mib[2] = SWPENC_ENABLE;
+  len = sizeof (swapenc);
+  if (sysctl(mib, 3, &swapenc, &len, NULL, 0) != -1)
+    if (swapenc != 0)
+      return;
+#endif
   if (!no_warning)
     {
       log_info(_("WARNING: using insecure memory!\n"));
